flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/AggregateUtil.scala

```diff
@@ -712,16 +712,14 @@ object AggregateUtil extends Enumeration {
     }
 
     // count(*) not exist in aggregateCalls, insert a count(*) in it.
-    val typeFactory = new FlinkTypeFactory(new FlinkTypeSystem)
     if (indexOfCountStar.isEmpty) {
-
       val count1 = AggregateCall.create(
         SqlStdOperatorTable.COUNT,
         false,
         false,
         new util.ArrayList[Integer](),
         -1,
-        typeFactory.createFieldTypeFromLogicalType(new BigIntType()),
+        new BasicSqlType(FlinkTypeSystem.INSTANCE, SqlTypeName.BIGINT),
```

Comment:
This is not possible. Types need to be `canonized` and need to go through the type factory once in order to allow reference equality.

Reply:
Are you sure? I just tried this on the debugger:

```
new BasicSqlType(FlinkTypeSystem.INSTANCE(), SqlTypeName.BIGINT)
  .equals(flinkTypeFactory.createFieldTypeFromLogicalType(new BigIntType().copy(false))
```

And It's true. 

I also debugged `canonize` and that function for `BasicSqlType` is always no-op, the `canonize` function is doing something only for `ObjectSqlType`